include/actor_types.h

```diff
@@ -5,32 +5,29 @@
 // Made this because there will likely be a lot of structs in this file that will be compiled into one eventually.
 // This keeps the other files cleaner.
 
+/*
+Note that while the `flag` members are signed, they should be understood as just a group of bits.
+Based on some googling around (see https://stackoverflow.com/a/11644749) the operation `thing.flag |= 0x8000`
+is NOT undefined behaviour, even though that is setting the sign bit.
+The numerical interpretation of the result as a signed number is implementation defined, while the
+actual operation acts on the bit representation of the number.
+*/
+
 struct piranha_plant {
-    /* 0x00 */ s16 unk0; // 0xC, 0xD, or 0x2B
-    #ifdef AVOID_UB
-    /* 0x02 */ u16 unk2;
-    #else
-               s16 unk2; // bitflag
-    #endif
-    /* 0x04 */ s16 unk4; // Timer?
-    /* 0x06 */ s16 unk6; // 1
-    /* 0x08 */ s16 unk8;
-    /* 0x0A */ s16 unkA;
-    /* 0x0C */ f32 unkC;
-    /* 0x10 */ Vec3s unk10; // Actor rotation
-    /* 0x18 */ s32 unk18;
-    /* 0x1C */ f32 unk1C;
-    /* 0x20 */ s16 unk20; 
-    /* 0x22 */ s16 unk22;
-    /* 0x24 */ s16 unk24;
-    /* 0x26 */ s16 unk26;
-    /* 0x28 */ s16 unk28;
-    /* 0x2A */ s16 unk2A;
+    /* 0x00 */ s16 type;
+    /* 0x02 */ s16 flags;
+    /* 0x04 */ Vec4s visibilityStates; // A per-camera visibilty state tracker
+    /* 0x0C */ f32 boundingBox;
```

Comment:
Yeah I was considering that as I put the PR up. The `kart_attributes` file(s) have a `gKartBoundingBoxTable` entry and that's where I got the naming scheme from. But it would more accurate to say that that table tracks the size of the bounding box rather being the bounding box itself.